component/src/main/java/org/wso2/extension/siddhi/store/ignite/ApacheIgniteConditionVisitor.java

```diff
@@ -0,0 +1,311 @@
+    }
+
+    public SortedMap<Integer, Object> getParametersConstant() {
+        return this.parametersConstant;
+    }
+
+    @Override
+    public void beginVisitAnd() {
+        condition.append(ApacheIgniteConstants.OPEN_PARENTHESIS);
+    }
+
+    @Override
+    public void endVisitAnd() {
+        condition.append(ApacheIgniteConstants.CLOSE_PARENTHESIS);
+    }
+
+    @Override
+    public void beginVisitAndLeftOperand() { }
+
+    @Override
+    public void endVisitAndLeftOperand() { }
+
+    @Override
+    public void beginVisitAndRightOperand() {
+        condition.append(ApacheIgniteConstants.SQL_AND).append(ApacheIgniteConstants.WHITESPACE);
+    }
+
+    @Override
+    public void endVisitAndRightOperand() { }
+
+    @Override
+    public void beginVisitOr() {
+        condition.append(ApacheIgniteConstants.OPEN_PARENTHESIS);
+    }
+
+    @Override
+    public void endVisitOr() {
+        condition.append(ApacheIgniteConstants.CLOSE_PARENTHESIS);
+    }
+
+    @Override
+    public void beginVisitOrLeftOperand() { }
+
+    @Override
+    public void endVisitOrLeftOperand() { }
+
+    @Override
+    public void beginVisitOrRightOperand() {
+        condition.append(ApacheIgniteConstants.SQL_OR).append(ApacheIgniteConstants.WHITESPACE);
+    }
+
+    @Override
+    public void endVisitOrRightOperand() { }
+
+    @Override
+    public void beginVisitNot() {
+        condition.append(ApacheIgniteConstants.SQL_NOT).append(ApacheIgniteConstants.WHITESPACE);
+    }
+
+    @Override
+    public void endVisitNot() { }
+
+    @Override
+    public void beginVisitCompare(Compare.Operator operator) {
+        condition.append(ApacheIgniteConstants.OPEN_PARENTHESIS);
+    }
+
+    @Override
+    public void endVisitCompare(Compare.Operator operator) {
+        condition.append(ApacheIgniteConstants.CLOSE_PARENTHESIS);
+    }
+
+    @Override
+    public void beginVisitCompareLeftOperand(Compare.Operator operator) { }
+
+    @Override
+    public void endVisitCompareLeftOperand(Compare.Operator operator) { }
+
+    @Override
+    public void beginVisitCompareRightOperand(Compare.Operator operator) {
+        switch (operator) {
+            case EQUAL:
+                condition.append(ApacheIgniteConstants.SQL_COMPARE_EQUAL);
+                break;
+            case GREATER_THAN:
+                condition.append(ApacheIgniteConstants.SQL_COMPARE_GREATER_THAN);
+                break;
+            case GREATER_THAN_EQUAL:
+                condition.append(ApacheIgniteConstants.SQL_COMPARE_GREATER_THAN_EQUAL);
+                break;
+            case LESS_THAN:
+                condition.append(ApacheIgniteConstants.SQL_COMPARE_LESS_THAN);
+                break;
+            case LESS_THAN_EQUAL:
+                condition.append(ApacheIgniteConstants.SQL_COMPARE_LESS_THAN_EQUAL);
+                break;
+            case NOT_EQUAL:
+                condition.append(ApacheIgniteConstants.SQL_COMPARE_NOT_EQUAL);
+                break;
```

Comment:
add a default case and throw an exception if other unsupported cases can be passed into this function